### PR TITLE
feat: delist item

### PR DESCRIPTION
**Pull request by AlefrankM**

### CODE REVIEW

**Review by GimerStackIt — DISMISSED** [review]
No comments

**Review by juliswer — DISMISSED** [review]
Looks good. Good job man

File: src/items/dto/delist-item-request.dto.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { IsNotEmpty } from 'class-validator';
+
+export class DelistItemRequestDto {
+  @IsNotEmpty()
```
Comment (mmiranda-stackit) [inline src/items/dto/delist-item-request.dto.ts:5]: Could you add the ethereum check ?

File: src/items/services/item.service.ts [hunk 2]
```diff
@@ -101,4 +102,24 @@ export class ItemService {
...
     return item;
   }
+
+  async delistAnItem({ itemId, address }: DelistItemRequestDto): Promise<Item> {
+    const item = await this.itemRepository.findById(itemId);
+    if (!item) throw new NotFoundException(`The item with id ${itemId} does not exist`);
+
+    await this.itemRepository.delistAnItem(itemId);
+
+    const account = await this.accountRepository.findByAddress(address);
```
Comment (mmiranda-stackit) [inline src/items/services/item.service.ts:111]: Move this up, before delisting. You have to make sure first if the account exist

File: src/items/services/item.service.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ import { HistoryRepository } from '../repositories/history.repository';
 import { TransactionType } from '../enums/transaction-type.enum';
```
Comment (mmiranda-stackit) [inline src/items/services/item.service.ts:10]: Add unit tests for the method

File: src/items/controllers/item.controller.ts [hunk 4]
```diff
@@ -42,18 +43,30 @@ export class ItemsController {
...
   listAnItem(
     @Param('itemId') itemId: string,
     @Body() listItemRequestDto: ListItemRequestDto
   ): Promise<Item> {
     listItemRequestDto.itemId = itemId;
     return this.itemService.listAnItem(listItemRequestDto);
   }
+
+  @IsAddressValid()
+  @Post('/:itemId/delist')
```
Comment (mmiranda-stackit) [inline src/items/controllers/item.controller.ts]: Should this be a patch?

**Review by mmiranda-stackit — CHANGES_REQUESTED** [review]

File: src/items/__tests__/item.service.spec.ts [hunk 5]
```diff
@@ -224,4 +228,33 @@ describe('ItemService', () => {
       });
     });
   });
+
+  describe('When delistAnItem function is called ', () => {
```
Comment (mmiranda-stackit) [inline src/items/__tests__/item.service.spec.ts:232]: What an amazing test.. Where are you learning this crap? :P

**Review by mmiranda-stackit — APPROVED** [review]
cool

File: src/items/__tests__/item.service.spec.ts (hunk 5 above)
Comment (AlefrankM) [reply src/items/__tests__/item.service.spec.ts:232]: O.o

**Pull request merged**